agreement/models/agreement.py

```diff
@@ -0,0 +1,84 @@
+# © 2017 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import _, api, fields, models
+
+
+class Agreement(models.Model):
+    _name = "agreement"
+    _description = "Agreement"
+    _inherit = ["mail.thread", "mail.activity.mixin"]
+
+    code = fields.Char(required=True, tracking=True)
+    name = fields.Char(required=True, tracking=True)
+    partner_id = fields.Many2one(
+        "res.partner",
+        string="Partner",
+        ondelete="restrict",
+        domain=[("parent_id", "=", False)],
+        tracking=True,
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        default=lambda self: self.env.company,
+    )
+    is_template = fields.Boolean(
+        string="Is a Template?",
+        default=False,
+        copy=False,
+        help="Set if the agreement is a template. "
+        "Template agreements don't require a partner.",
+    )
+    agreement_type_id = fields.Many2one(
+        "agreement.type",
+        string="Agreement Type",
+        help="Select the type of agreement",
+    )
+    domain = fields.Selection(
+        "_domain_selection",
+        default="sale",
+        tracking=True,
+    )
+    active = fields.Boolean(default=True)
+    signature_date = fields.Date(tracking=True)
+    start_date = fields.Date(tracking=True)
+    end_date = fields.Date(tracking=True)
+
+    @api.model
+    def _domain_selection(self):
+        return [
+            ("sale", _("Sale")),
+            ("purchase", _("Purchase")),
+        ]
+
+    @api.onchange("agreement_type_id")
+    def agreement_type_change(self):
```

Comment:
I suggest to change domain field into computed store=True, readonly=False one as onchange tend to disappear in Odoo

Reply:
Done in a separated commit.